ref: Restructure event payload docs (#1166)
getsentry/sentry-docs: src/collections/_documentation/development/sdk-dev/event-payloads/contexts.md

---
title: Contexts Interface
sidebar_order: 9
---

The context interfaces provide additional context data. Typically this is data
related to the current user, the current HTTP request. Its canonical name is
`"contexts"`.

> The `contexts` type can be used to defined almost arbitrary contextual data on

Author: Yes, good point. It's better without "almost"!

```suggestion
The `contexts` type can be used to define arbitrary contextual data on
```

the event. It accepts an object of key, value pairs. The key is the “alias” of
the context and can be freely chosen. However as per policy it should match the

> type of the context unless there are two values for a type. `type` can be

Author: ```suggestion
type of the context unless there are two values for a type. You can omit `type`
```

: _Optional_. Free device storage in bytes.

`external_storage_size`:

: _Optional_. Total size of an attached external storage in bytes (e.g.: android
  SDK card).

`external_free_storage`:

: _Optional_. Free size of an attached external storage in bytes (e.g.: android
  SDK card).

`boot_time`:

: _Optional_. A formatted UTC timestamp when the system was booted, e.g.:
  `"2018-02-08T12:52:12Z"`.

`timezone`:

: _Optional_. The timezone of the device, e.g.: `Europe/Vienna`.

## OS Context

Describes the operating system on which the event was created. In web
contexts, this is the operating system of the browser (normally pulled from
the User-Agent string).

`name`:

: _Optional_. The name of the operating system.

`version`:

: _Optional_. The version of the operating system.

`build`:

: _Optional_. The internal build revision of the operating system.

`kernel_version`:

: _Optional_. An independent kernel version string. This is typically the
  entire output of the `uname` syscall.

`rooted`:

: _Optional_. A flag indicating whether the OS has been jailbroken or rooted.

`raw_description`:

: _Optional_. An unprocessed description string obtained by the operating

>   system. For some well-known runtimes, Sentry will attempt to parse `name`

Author: We don't want to commit here, which we will parse. Well-known in this context effectively means macOS, Windows, and certain Linux distros, but we're constantly updating.

  and `version` from this string, if they are not explicitly given.

## Runtime Context

Describes a runtime in more detail. Typically this context is used multiple
times if multiple runtimes are involved (for instance if you have a JavaScript
application running on top of JVM)

`name`:

: _Optional_. The name of the runtime.

`version`:

: _Optional_. The version identifier of the runtime.

`raw_description`:

: _Optional_. An unprocessed description string obtained by the runtime. For
  some well-known runtimes, Sentry will attempt to parse `name` and `version`
  from this string, if they are not explicitly given.

## App Context

Describes the application. As opposed to the runtime, this is the actual
application that was running and carries meta data about the current session.

`app_start_time`:

: _Optional_. Formatted UTC timestamp when the application was started by the
  user.

`device_app_hash`:

> : _Optional_. Application specific device identifier.

Reviewer: ```suggestion
: _Optional_. Application-specific device identifier.
```

`build_type`:

> : _Optional_. String identifying the kind of build, e.g. `testflight`.

Reviewer: ```suggestion
: _Optional_. String identifying the kind of build. For example, `testflight`.
```

`app_identifier`:

: _Optional_. Version-independent application identifier, often a dotted
  bundle ID.

`app_name`:

: _Optional_. Human readable application name, as it appears on the platform.

`app_version`:

: _Optional_. Human readable application version, as it appears on the
  platform.

`app_build`:

: _Optional_. Internal build identifier, as it appears on the platform.

## Browser Context

> Carries information about the browser or user agent for web-related errors.

Reviewer: ```suggestion
Browser context carries information about the browser or user agent for web-related errors.
```

> This can either be the browser this event ocurred in, or the user agent of a

Reviewer: ```suggestion
This can either be the browser this event occurred in or the user agent of a
```

web request that triggered the event.

Attributes:

`name`:

: _Optional_. Display name of the browser application.

`version`:

: _Optional_. Version string of the browser.

## GPU Context

> Describes the GPU of the device.

Reviewer: ```suggestion
GPU context describes the GPU of the device.
```

`name`:

: **Required**. The name of the graphics device.

`version`:

> : _Optional_. The Version of graphics device.

Reviewer: ```suggestion
: _Optional_. The Version of the graphics device.
```

`id`:

: _Optional_. The PCI identifier of the graphics device.

`vendor_id`:

: _Optional_. The PCI vendor identifier of the graphics device.

`vendor_name`:

: _Optional_. The vendor name as reported by the graphics device.

`memry_size`:

: _Optional_. The total GPU memory available in Megabytes.

`api_type`:

> : _Optional_. The device low level API type.

Reviewer: ```suggestion
: _Optional_. The device low-level API type.
```
